.github/workflows/build.yaml

```diff
@@ -64,41 +64,81 @@ jobs:
              make_args="${make_args} image=${{ env.image }}"
          fi
          echo "make_args=${make_args}" >> $GITHUB_ENV
-         c_image_host=$(echo ${{ env.image }} | cut -d '/' -f 1 | rev | cut -d '.' -f 1-2 | rev)
-         case ${c_image_host} in
-             docker.com | docker.io)
-                 d_registry="docker.io"
-                 d_username=${{ secrets.DOCKER_USERNAME }}
-                 d_password=${{ secrets.DOCKER_TOKEN }}
-                 ;;
-             ghcr.io)
-                 d_registry="ghcr.io"
-                 d_username="token"
-                 d_password=${{ secrets.GHCR_PUSH_TOKEN}}
-                 ;;
-             *)
-                 d_registry=${c_image_host}
-                 d_username=''
-                 d_password=''
-                 ;;
-         esac
-         echo "d_registry=${d_registry}" >> $GITHUB_ENV
-         echo "d_username=${d_username}" >> $GITHUB_ENV
-         echo "d_password=${d_password}" >> $GITHUB_ENV
+         split_images=$(echo ${{ env.image}} | tr ',' ' '
+         for img in ${split_images}; do
+             image_host=$(echo ${img} | cut -d '/' -f 1)
+             # Check for implicit Docker Hub
+             case ${image_host} in
+                 *.*)
+                     ;;
+                 *)
+                     image_host="docker.io"
+                     ;;
+             esac
+             case ${image_host} in
+                 docker.com | docker.io) | *.docker.com | *.docker.io)
+                     d_tag="docker_"
+                     d_registry="docker.io"
+                     d_username=${{ secrets.DOCKER_USERNAME }}
+                     d_password=${{ secrets.DOCKER_TOKEN }}
+                     ;;
+                 ghcr.io | *.ghcr.io)
+                     d_tag="github_"
+                     d_registry="ghcr.io"
+                     d_username="token"
+                     d_password=${{ secrets.GHCR_PUSH_TOKEN}}
+                     ;;
+                 *-docker.pkg.dev)
+                     d_tag="google_"
+                     d_registry=${image_host}
+                     d_username="_json_key"
+                     d_password=${{ secrets.GAR_PUSH_TOKEN }}
+                 *)
+                     d_tag=""
+                     d_registry=${image_host}
+                     d_username=""
+                     d_password=""
+                     ;;
+             esac
+             echo "${d_tag}registry=${d_registry}" >> $GITHUB_ENV
+             echo "${d_tag}username=${d_username}" >> $GITHUB_ENV
+             echo "${d_tag}password=${d_password}" >> $GITHUB_ENV
 
-    - name: Log in to Docker repository
-      # Only bother if we're going to push the container and we have
-      #  credentials
+    # Only bother logging in to a given site if we're going to push the
+    # container and we have credentials
+    - name: Log in to Docker Hub repository
```

Comment:
It's a shame there is so much code duplication here.

Reply:
In theory I could do it with a matrix, but that would be even more confusing.